app/components/agenda-manager/agenda-context.hbs

```diff
@@ -1,8 +1,9 @@
 {{yield (hash
```

Comment:
Unrelated to these issues but I wanted to get the discussion started:

Personally I don't really see a benefit of the AgendaContext component. I understand the intention but in this case it only seems to add unnecessary indirection when trying to understand how the "AgendaManager" works. I think extracting that logic back into the AgendaManager component itself makes the most sense and would make things simpler and easier to follow.

Reply:
I guess the hope was that you don't need to understand how the context methods work and just use them as a blackbox api
but I don't think I quite managed that with this code

Reply:
Yea I understand your reasoning. I like that pattern as well, especially if it needs to be reusable. [I've used it recently myself](https://github.com/lblod/ember-rdfa-helpers/pull/8/files#diff-179a4536ca2838b76d93bda0f10394dd9c19f29ff2332505904c3c7adf6ccbcdR1-R8) to refactor away from a mixin.

Now that I look at it again, I think the thing I dislike about the current implementation is that it passes the `createAgendaItem` task as a param to the `createItemTask` on the parent component. It's harder to follow how everything works imo. The thing I like more is to pass the parent task as an argument to the context component, which can tell call it when needed (which is what I did in this PR for the reset functionality). I think that flow is easier to follow and more in line with how you interact with other components.

Anyway, it probably won't be worth the energy to remove it or change it since everything works already and it's just my opinion. So ignore my ramblings. :smile:

Reply:
ah yes so something like an onCreate argument on the context rather than the other way round? I agree that makes more sense, I wasn't happy with my solution either I recall

Reply:
I added the change since it was a small fix anyway.